crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -289,27 +224,85 @@ impl AlgorithmUpdaterV1 {
 
     fn update_exec_gas_price(&mut self, used: u64, capacity: NonZeroU64) {
         let threshold = *self.l2_block_fullness_threshold_percent as u64;
-        let mut exec_gas_price = self.new_scaled_exec_price;
+        let mut scaled_exec_gas_price = self.new_scaled_exec_price;
         let fullness_percent = used
             .saturating_mul(100)
             .checked_div(capacity.into())
             .unwrap_or(threshold);
 
         match fullness_percent.cmp(&threshold) {
             std::cmp::Ordering::Greater => {
-                let change_amount = self.change_amount(exec_gas_price);
-                exec_gas_price = exec_gas_price.saturating_add(change_amount);
+                let change_amount = self.exec_change(scaled_exec_gas_price);
+                scaled_exec_gas_price =
+                    scaled_exec_gas_price.saturating_add(change_amount);
             }
             std::cmp::Ordering::Less => {
-                let change_amount = self.change_amount(exec_gas_price);
-                exec_gas_price = exec_gas_price.saturating_sub(change_amount);
+                let change_amount = self.exec_change(scaled_exec_gas_price);
+                scaled_exec_gas_price =
+                    scaled_exec_gas_price.saturating_sub(change_amount);
             }
             std::cmp::Ordering::Equal => {}
         }
-        self.new_scaled_exec_price = max(self.min_exec_gas_price, exec_gas_price);
+        self.new_scaled_exec_price =
+            max(self.min_scaled_exec_gas_price(), scaled_exec_gas_price);
+    }
+
+    fn min_scaled_exec_gas_price(&self) -> u64 {
+        self.min_exec_gas_price
+            .saturating_mul(self.gas_price_factor.into())
+    }
+
+    fn update_da_gas_price(&mut self) {
+        let p = self.p();
+        let d = self.d();
+        let da_change = self.da_change(p, d);
+        let maybe_new_scaled_da_gas_price = (self.new_scaled_da_gas_price as i128)
+            .checked_add(da_change)
+            .and_then(|x| u64::try_from(x).ok())
+            .unwrap_or(if da_change.is_positive() {
+                u64::MAX
+            } else {
+                0u64
+            });
```

Comment:
I think it's worth considering. My concern is more about being explicit about why we are clamping it to those values. For example, above, we are
```rs
            let projected_total_da_cost =
                i128::try_from(self.projected_total_da_cost).unwrap_or(i128::MAX);
```
and it's not exactly clear why we're okay with truncating `u128::MAX` to `i128::MAX`. Or give us a great chance to have a conversation about if that's okay.

I'm more worried about defining the behavior than I am about DRYing the code.

Reply:
I created a helper/getter for `projected_total_da_cost`

Reply:
Yeah ideally we shouldn't be near the boundaries of having to clamp these values at all either. It would be very strange if the controller even got close to suggesting such a large change :sweat_smile: